### PR TITLE
Fix: generator-star-spacing with class methods (fixes #2351)

### LINKED ISSUE #2351: generator-star-spacing support for methods in classes
The following configuration works well for both function and methods in object literals but fails for methods in classes:

```json
{
  "env": {
    "es6": true
  },

  "rules": {
    "generator-star-spacing": [2, "after"]
  }
}
```

Errors are not detected for methods in classes:

```javascript
class className {
  *methodName() {}
}
```

But they are for functions and methods in object literals:

```javascript
function*funcName() {}

({
  *methodName() {}
})
```

### PR DESCRIPTION
**Pull request by btmills**

### CODE REVIEW

File: lib/rules/generator-star-spacing.js [hunk 1]
```diff
@@ -43,20 +43,27 @@ module.exports = function(context) {
     }
 
     /**
+     * Checks whether the current node is a class or object method.
+     * @returns {boolean} True if the current node is a method.
+     */
+    function isMethod() {
+        var parent = context.getAncestors().pop();
```
Comment (nzakas) [inline lib/rules/generator-star-spacing.js]: Probably better to pass in the node and use `node.parent` to make sure you have the right data.

**Comment by michaelficarra** [comment]
Can you add tests for static class methods?

**Comment by btmills** [comment]
Done.

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**